src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs

```diff
@@ -2108,5 +2108,190 @@ private void ReportDiagnosticsCore(Action<Diagnostic> addDiagnostic, Location lo
                 addDiagnostic(diagnostic);
             }
         }
+
+        [Fact, WorkItem(23309, "https://github.com/dotnet/roslyn/issues/23309")]
+        public void TestFieldReferenceAnalyzer_InAttributes()
+        {
+            string source = @"
+using System;
+
+[assembly: MyAttribute(C.FieldForAssembly)]
```

Comment:
Also updated C# and VB unit tests to verify callbacks for attributes on assembly, module, type parameter, parameter and return type.